umap/static/umap/js/umap.components.js

```diff
@@ -0,0 +1,13 @@
+class UmapFragment extends HTMLElement {
+  connectedCallback() {
+    new L.U.Map(this.firstElementChild.id, JSON.parse(this.dataset.settings))
```

Comment:
I suppose the reason the id is in the inner `div` (rather than directly in the `umap-fragment`) is that umap currently expect a `div`element? 🤔

Reply:
Yes, `Map()` is expecting an `id` for the container and I reused the already defined `class` to avoid a bigger change. Somehow we have a cleaner separation between data (settings) related to the component and id/class for the rendering of the map.

One option could have been to generate the `div` from `umap-fragment` parameters but this way we have the HTML part in an HTML file :)